python/hidet/ir/compute/cops/reduce.py

```diff
@@ -0,0 +1,56 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List
+from hidet.ir.compute import reduce
+from hidet.graph.ops.utils import TensorNode
+from hidet.graph.ops.utils import compute
+
+
+def reduce_shape(x: TensorNode, dims: List[int], keep_dim: bool):
+    y_shape = []
+    dims = [i if i >= 0 else len(x.shape) + i for i in dims]
+    for i in range(len(x.shape)):
+        if i in dims:
+            if keep_dim:
+                y_shape.append(1)
+        else:
+            y_shape.append(x.shape[i])
+    return y_shape
+
+
+def reduce_cop(x: TensorNode, dims: List[int], keep_dim: bool, reduce_type: str, accumulate_dtype: str = 'float32'):
```

Comment:
We do not need to add the `_cop` suffix to the compute operators as we will always place them in the `cops` namespace. Thus, when we use them, we can code like
```
from hidet.ir.compute import cops
cops.reduce(...)
cops.matmul(...)
```
without worrying the name conflicts (say with the ones under hidet.ops).